### PR TITLE
Centralize IP address types and utilities

### PR DESCRIPTION
**Pull request by fhunleth**

### CODE REVIEW

File: lib/vintage_net/ip.ex [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+      iex> VintageNet.IP.ip_to_string("192.168.9.1")
+      "192.168.9.1"
+
+      iex> VintageNet.IP.ip_to_string({65152, 0, 0, 0, 0, 0, 0, 1})
+      "fe80::1"
+  """
+  @spec ip_to_string(VintageNet.any_ip_address()) :: String.t()
+  def ip_to_string(ipa) when is_tuple(ipa) do
+    :inet.ntoa(ipa) |> to_string()
```
Comment (mattludwigs) [inline lib/vintage_net/ip.ex]: `:inet.ntoa/1` can return `{:error, :einval}`, not sure if we want to handle that as the type spec should catch any issues with incoming `ipa` formats. But that assumes the consumer of `vintage_net` is using dialyzer.

I am assuming ignoring that case is relatively safe, but just wanted to make sure that was on purpose.

Comment (fhunleth) [reply lib/vintage_net/ip.ex]: Good catch. I'll address shortly...

Comment (fhunleth) [reply lib/vintage_net/ip.ex]: Fixed. I added a unit test to make sure this crashes. This particular function is more internal than external. However, there's another one that converts externally produced IP addresses and I beefed that one up (and added tests) to give users a nice message if they pass bad IP addresses into VintageNet.

Comment (mattludwigs) [reply lib/vintage_net/ip.ex]: :+1:

**Pull request merged**